src/I_BatteryStateOfChargeService.h

```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 #include <QTime>
-struct BatteryData;
+#include <BatteryData.h>
```

Comment:
So the `struct BatteryData` here was fine, this is called forward declaring it, don't worry about what that means for now. But having the full include `#include <BatteryData.h>` doesn't need to be in this file. You should have the full include in the cpp file for BatteryStateOfChargeService.